File: historyscan-back/src/main/java/io/sch/historyscan/domain/contexts/codebase/find/CodeBaseInfoInventory.java
package io.sch.historyscan.domain.contexts.codebase.find;

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureSPI;

import java.util.Optional;

@HexagonalArchitectureSPI
@FunctionalInterface
public interface CodeBaseInfoInventory {
    Optional<CurrentCodeBase> findBy(String name);
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/codebase/find/CodeBaseInfoInventory.kt
package io.sch.historyscan.domain.contexts.codebase.find

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureSPI

@HexagonalArchitectureSPI
fun interface CodeBaseInfoInventory {
    fun findBy(name: String): CurrentCodeBase?
}
